airflow/api_connexion/endpoints/dag_endpoint.py

```diff
@@ -81,11 +82,21 @@ def get_dag_details(
     for key, value in dag.__dict__.items():
         if not key.startswith("_") and not hasattr(dag_model, key):
             setattr(dag_model, key, value)
+
+    serialized_dag = session.scalar(select(SerializedDagModel).where(SerializedDagModel.dag_id == dag_id))
+    dataset_expression = {}
+    if serialized_dag:
+        dataset_expression = serialized_dag.get_dataset_expression()
```

Comment:
```suggestion
    if serialized_dag := session.scalar(select(SerializedDagModel).where(SerializedDagModel.dag_id == dag_id)):
        dataset_expression = serialized_dag.get_dataset_expression()
    else:
        dataset_expression = {}
```

if this is not too long